### ISSUE TITLE
[Web] flutter app is accessible when uses JAWS

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#75881]
JAWS is the windows only screen reader https://www.freedomscientific.com/products/software/jaws/

We need to make sure our flutter web application on windows browser is accessible through JAWS

**Comment by ferhatb** [comment]
@chunhtai , is there an open issue or is there a task associated?

**Comment by chunhtai** [comment]
There is https://github.com/flutter/flutter/issues/75879 to track all the new rating requirements. This is an investigating issue to track that we have tested flutter can work with JAWS.

**Comment by chunhtai** [comment]
Hi @yjbanov I remembered you were working on this, should i assign this to you?

**Comment by BerndWessels** [comment]
using JAWS with flutter web app seems pretty broken.

navigation with arrow keys kind of announces most of the semantic widgets but tab navigation is broken and pressing enter or space on buttons for example doesn't work at all.

In my opinion flutter for web is not accessible at all at the moment and therefore pretty much a no go for any serious web application.

I hope this is just a bug and can be addressed easily.

Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.0.6, on Microsoft Windows [Version 10.0.19042.928], locale en-NZ)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code, 64-bit edition (version 1.56.1)
[√] Connected device (2 available)

**Comment by TahaTesser** [comment]
I can reproduce the issue, JAWS doesn't work on the web 
<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel master, 2.3.0-2.0.pre.254, on Microsoft Windows [Version 10.0.19042.964], locale en-US)
    • Flutter version 2.3.0-2.0.pre.254 at C:\Users\Taha\Code\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e716ac19e8 (5 hours ago), 2021-05-18 00:04:02 -0700
    • Engine revision 3cda7f5807
    • Dart version 2.14.0 (build 2.14.0-119.0.dev)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\Taha\Code\SDK
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Users\Taha\Code\android-studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
    • All Android licenses accepted.
[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe
[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.5)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.9.31229.75
    • Windows 10 SDK version 10.0.19041.0
[✓] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
[✓] VS Code (version 1.56.2)
    • VS Code at C:\Users\Taha\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.22.0
[✓] Connected device (4 available)
    • IN2011 (mobile)   • c9d8ee0c • android-arm64  • Android 11 (API 30)
    • Windows (desktop) • windows  • windows-x64    • Microsoft Windows [Version 10.0.19042.964]
    • Chrome (web)      • chrome   • web-javascript • Google Chrome 90.0.4430.212
    • Edge (web)        • edge     • web-javascript • Microsoft Edge 89.0.774.63
• No issues found!
```

</details>

**Comment by BerndWessels** [comment]
anybody looking at this? and as a bonus question, can we trust that Flutter Web is accessible in the near future?

**Comment by ErizoDaFunk** [comment]
Actually, Im working on an App, and a requirement of accesibilty has been recently added. Does somebody know if it is possible to develop a web app on flutter correctly usable by JAWs or should we restart the app on react (for example)?

**Comment by hapasa** [comment]
I wrote an app with is more than one page, with a button in AppBar to go to a settings page done with flutter_settings_screens.

Building it for Chrome and trying to use it with VoiceOver on a Mac using keyboard navigation was pretty much a failure.   Still playing with Sematics to see if I can make it better.    It did work a little bit better when targeting Web than when targeting MacOS.   I did get the impression that with a touch screen it could be more usable.

Despite adding Semantics widget wrappers, MacOS stayed unusable.  

Other notes:
The SettingsScreen is totally inaccessible out of the box, I'll try to see if adding some Semantics there would help. 
Flutter Slider is not usable from keyboard.

**Comment by BerndWessels** [comment]
I am raising a critical accessibility concern regarding JAWS and VoiceOver compatibility with Flutter web applications since there has been no visible progress. Current limitations in keyboard navigation and screen reader interaction pose a significant barrier for users with disabilities.

Impact:

- Limited Accessibility: Users relying on JAWS and VoiceOver may struggle to navigate and interact with Flutter web apps. This could violate WCAG guidelines and potentially lead to legal issues related to digital accessibility discrimination.
- Documentation Gap: The official Flutter documentation does not currently highlight these limitations.

Proposed Action:

To ensure inclusivity and mitigate legal risks, I urge the Flutter team to:

- Prioritize Accessibility: Dedicate resources to improving compatibility with JAWS and VoiceOver in Flutter web apps.
- Update Documentation: Clearly state the current limitations in the official documentation, informing developers of the potential lack of accessibility for users relying on specific screen readers.

By addressing these concerns, we can ensure that Flutter web apps are inclusive and accessible to everyone.

Additional Notes:

WCAG guidelines (https://www.w3.org/WAI/standards-guidelines/wcag/) and potential legal implications of inaccessible websites (https://www.forbes.com/sites/gusalexiou/2023/06/30/website-accessibility-lawsuits-rising-exponentially-in-2023-according-to-latest-data/).

**Comment by yjbanov** [comment]
JAWS is supported. I think this umbrella issue can be closed. For specific bugs or feature requests let's open individual issues and address them one at a time.
